### ISSUE TITLE
No hook (i.e. none of selected, autocomplete or cursorchange fired) if user types in exact match

### ISSUE THREAD

**Issue by aholland** [corejavascript/typeahead.js#33]
I have bloodhound configured to fetch Json data, which it does very nicely. 
<pre>
   // constructs the suggestion engine
    var bloodHound = new Bloodhound({
        datumTokenizer: Bloodhound.tokenizers.whitespace,
        queryTokenizer: Bloodhound.tokenizers.whitespace,
        remote: {
            url: 'http://localhost:9000/invoices%QUERY',
            wildcard: '%QUERY'
        }
    });
</pre>

In addition, the three main custom events are bound to the function "selectedHandler".
<pre>
    ta.bind("typeahead:cursorchange", selectedHandler);
    ta.bind("typeahead:autocomplete", selectedHandler);
    ta.bind("typeahead:select", selectedHandler);
</pre>

So almost everything works: if the user types in a string that matches part of a result and completes using tab, the autocomplete event fires and the correct item is selected. If the user uses the arrow keys or the mouse to select from the dropdown, that works. Even if the user type in the whole string matching a result but in the wrong case, and then presses tab, that works. BUT if the user types in the exact string of the result, no event is fired and there is no direct way that I can find to get the data from the matching Json result into the form. I think this is a bug.

**Comment by mallyvai** [comment]
I want to second this as being an issue that bites me as well. Was there a resolution? I see https://github.com/corejavascript/typeahead.js/pull/34 referenced here - is it a satisfactory fix? What do the maintainers think?

**Comment by aholland** [comment]
Vaibhav, if you're asking me: I'm afraid I've heard nothing. But I'm happy with my patch for now. One thing, I did become aware of yet another fork, https://github.com/algolia/autocomplete.js

**Comment by jcrben** [comment]
I agree that this would be a bug - @mallyvai have you checked out #34? If so please comment on the pull request. The maintainers are still getting up to speed, and at least I am not exactly an expert in the library.

**Comment by sileence** [comment]
Having the same problem here
